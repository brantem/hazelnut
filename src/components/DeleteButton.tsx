import { useState } from 'react';

type DeleteButtonProps = {
  onConfirm: () => void;
};

const DeleteButton = ({ onConfirm }: DeleteButtonProps) => {
  const [isDelete, setIsDelete] = useState(false);

  return !isDelete ? (
    <button
      className="rounded-md p-3 text-left hover:bg-neutral-100"
      onClick={() => {
        setIsDelete(true);
        setTimeout(() => setIsDelete(false), 2000);
      }}
    >
      Delete
    </button>
  ) : (
    <button className="rounded-md p-3 text-left text-red-500 hover:bg-red-50" onClick={onConfirm}>
      Click to Confirm
    </button>
  );
};

export default DeleteButton;
