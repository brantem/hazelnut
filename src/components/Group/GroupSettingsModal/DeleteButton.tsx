import { useState } from 'react';

import { useGroupsStore } from 'lib/stores';

type DeleteButtonProps = {
  groupId: string;
  onClick: () => void;
};

const DeleteButton = ({ groupId, onClick }: DeleteButtonProps) => {
  const { remove } = useGroupsStore();

  const [isDelete, setIsDelete] = useState(false);

  return !isDelete ? (
    <button
      className="rounded-md p-3 text-left hover:bg-neutral-100"
      onClick={() => {
        setIsDelete(true);
        setTimeout(() => setIsDelete(false), 2000);
      }}
    >
      Delete
    </button>
  ) : (
    <button
      className="rounded-md p-3 text-left text-red-500 hover:bg-red-50"
      onClick={() => {
        remove(groupId);
        onClick();
      }}
    >
      Click to Confirm
    </button>
  );
};

export default DeleteButton;
