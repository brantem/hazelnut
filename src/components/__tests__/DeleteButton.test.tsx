import { render, screen, act } from '@testing-library/react';
import '@testing-library/jest-dom';

import DeleteButton from 'components/DeleteButton';

describe('DeleteButton', () => {
  beforeEach(() => {
    vi.useFakeTimers();
  });

  afterEach(() => {
    vi.restoreAllMocks();
  });

  it('should render successfully', () => {
    const onConfirm = vi.fn(() => {});
    const { container } = render(<DeleteButton onConfirm={onConfirm} />);

    expect(container).toMatchSnapshot();
    const button = screen.getByText('Delete');
    expect(button).toBeInTheDocument();
    act(() => button.click());
    expect(screen.queryByText('Delete')).not.toBeInTheDocument();

    const confirm = screen.getByText('Click to Confirm');
    expect(confirm).toBeInTheDocument();
    act(() => confirm.click());
    expect(onConfirm).toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  it('should timed out', async () => {
    render(<DeleteButton onConfirm={() => {}} />);

    const button = screen.getByText('Delete');
    expect(button).toBeInTheDocument();
    act(() => button.click());
    expect(screen.queryByText('Delete')).not.toBeInTheDocument();
    expect(screen.getByText('Click to Confirm')).toBeInTheDocument();
    act(() => {
      vi.advanceTimersToNextTimer();
    });
    expect(screen.queryByText('Click to Confirm')).not.toBeInTheDocument();
    expect(screen.getByText('Delete')).toBeInTheDocument();
  });
});
