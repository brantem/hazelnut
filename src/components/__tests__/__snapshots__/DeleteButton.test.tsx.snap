// Vitest Snapshot v1

exports[`DeleteButton > should render successfully 1`] = `
<div>
  <button
    class="rounded-md p-3 text-left hover:bg-neutral-100"
  >
    Delete
  </button>
</div>
`;

exports[`DeleteButton > should render successfully 2`] = `
<div>
  <button
    class="rounded-md p-3 text-left text-red-500 hover:bg-red-50"
  >
    Click to Confirm
  </button>
</div>
`;
